src/libraries/PausableOwnable.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity >=0.8.8 <0.9.0;
+
+import "./Pausable.sol";
+import "./external/OwnableUpgradeable.sol";
+
+abstract contract PausableOwnable is PausableUpgradeable, OwnableUpgradeable {
+    /*
+     * @dev Modifier to allow only the Pauser and the Owner to access pausing functionality
+     */
+    modifier onlyOwnerOrPauser() {
+        _checkOwnerOrPauser(owner());
+        _;
+    }
+
+    /*
+     * @dev Modifier to allow only the Pauser to access some functionality
+     */
+    function _checkOwnerOrPauser(address owner) internal view {
+        if (pauser() != msg.sender && owner != msg.sender) {
+            revert InvalidPauser(msg.sender);
+        }
+    }
+
+    function __PausedOwnable_init(address initialPauser, address owner) internal onlyInitializing {
+        __Paused_init(initialPauser);
+        __Ownable_init(owner);
+    }
+
+    /**
+     * @dev Transfers the ability to pause to a new account (`newPauser`).
+     */
+    function _transferPauserCapability(address newPauser) internal virtual onlyOwnerOrPauser {
+        PauserStorage storage $ = _getPauserStorage();
+        address oldPauser = $._pauser;
+        $._pauser = newPauser;
+        emit PauserTransferred(oldPauser, newPauser);
+    }
+
+    /**
+     * @dev Leaves the contract without a Pauser
+     */
+    function _renouncePauser() public virtual onlyOwnerOrPauser {
```

Comment:
No leading underscore for external/public methods please.

Or alternatively if this is supposed to be internal then it should sit behind the `renouncePauser` virtual method...it's only defined in EndpointStandalone so far but should probably be defined here instead so it can be used by the manager and overriden if needed

Reply:
yea let's leave it public. removed the virtual method from EndpointStandalone. Manager directly call now.